plugins/mysql/mysqlembed.cpp

```diff
@@ -1020,17 +1012,96 @@ static void initializeMySqlThread()
+{
+    ParamTypeNone,
+    ParamTypeString,
+    ParamTypeUInt,
+    ParamTypeULong,
+    ParamTypeBool
+};
+
+struct MySQLOptionDefinition
+{
+    const char *name;
+    enum mysql_option option;
+    MySQLOptionParamType paramType;
+};
+
+#define addoption(a,b) { #a, a, b }
+
+MySQLOptionDefinition options[] =
+{
+  addoption(MYSQL_DEFAULT_AUTH, ParamTypeString),
+  addoption(MYSQL_ENABLE_CLEARTEXT_PLUGIN, ParamTypeBool),
+  addoption(MYSQL_INIT_COMMAND, ParamTypeString),
+  addoption(MYSQL_OPT_BIND, ParamTypeString),
+  addoption(MYSQL_OPT_CAN_HANDLE_EXPIRED_PASSWORDS, ParamTypeBool),
+  addoption(MYSQL_OPT_COMPRESS, ParamTypeNone),
+  addoption(MYSQL_OPT_CONNECT_ATTR_DELETE, ParamTypeString),
+  addoption(MYSQL_OPT_CONNECT_ATTR_RESET, ParamTypeNone),
+  addoption(MYSQL_OPT_CONNECT_TIMEOUT, ParamTypeUInt),
+  addoption(MYSQL_OPT_GUESS_CONNECTION, ParamTypeNone),
+  addoption(MYSQL_OPT_LOCAL_INFILE, ParamTypeUInt),
+  addoption(MYSQL_OPT_NAMED_PIPE, ParamTypeNone),
+#if (MYSQL_VERSION_ID >= 50709)
+  addoption(MYSQL_OPT_MAX_ALLOWED_PACKET, ParamTypeULong),
+  addoption(MYSQL_OPT_NET_BUFFER_LENGTH, ParamTypeULong),
+#endif
+  addoption(MYSQL_OPT_PROTOCOL, ParamTypeUInt),
+  addoption(MYSQL_OPT_READ_TIMEOUT, ParamTypeUInt),
+  addoption(MYSQL_OPT_RECONNECT, ParamTypeBool),
+  addoption(MYSQL_OPT_SSL_CA, ParamTypeString),
+  addoption(MYSQL_OPT_SSL_CAPATH, ParamTypeString),
+  addoption(MYSQL_OPT_SSL_CERT, ParamTypeString),
+  addoption(MYSQL_OPT_SSL_CIPHER, ParamTypeString),
+  addoption(MYSQL_OPT_SSL_CRL, ParamTypeString),
+  addoption(MYSQL_OPT_SSL_CRLPATH, ParamTypeString),
+#if (MYSQL_VERSION_ID >= 50703)
+  addoption(MYSQL_OPT_SSL_ENFORCE, ParamTypeBool),
+#endif
+  addoption(MYSQL_OPT_SSL_KEY, ParamTypeString),
+#if (MYSQL_VERSION_ID >= 50711)
+  addoption(MYSQL_OPT_SSL_MODE, ParamTypeUInt),
+#endif
+  addoption(MYSQL_OPT_SSL_VERIFY_SERVER_CERT, ParamTypeBool),
+#if (MYSQL_VERSION_ID >= 50710)
+  addoption(MYSQL_OPT_TLS_VERSION, ParamTypeString),
+#endif
+  addoption(MYSQL_OPT_USE_EMBEDDED_CONNECTION, ParamTypeNone),
+  addoption(MYSQL_OPT_USE_REMOTE_CONNECTION, ParamTypeNone),
+  addoption(MYSQL_OPT_USE_RESULT, ParamTypeNone),
+  addoption(MYSQL_OPT_WRITE_TIMEOUT, ParamTypeUInt),
+  addoption(MYSQL_PLUGIN_DIR, ParamTypeString),
+  addoption(MYSQL_READ_DEFAULT_FILE, ParamTypeString),
+  addoption(MYSQL_READ_DEFAULT_GROUP, ParamTypeString),
+  addoption(MYSQL_REPORT_DATA_TRUNCATION, ParamTypeBool),
+  addoption(MYSQL_SECURE_AUTH, ParamTypeBool),
+  addoption(MYSQL_SERVER_PUBLIC_KEY, ParamTypeString),
+  addoption(MYSQL_SET_CHARSET_DIR, ParamTypeString),
+  addoption(MYSQL_SET_CHARSET_NAME, ParamTypeString),
+  addoption(MYSQL_SET_CLIENT_IP, ParamTypeString),
+  addoption(MYSQL_SHARED_MEMORY_BASE_NAME, ParamTypeString),
+  { nullptr, (enum mysql_option) 0, ParamTypeNone }
+};
+
+static MySQLOptionDefinition &lookupOption(const char *optName)
+{
+    for (MySQLOptionDefinition *optDef = options; optDef->name != nullptr; optDef++)
```

Comment:
I wonder if it's worthwhile to contain the options in a map for lookup in order to avoid this iterative approach?